01-ruby-fundamentals/keyword-arguments.md

```diff
@@ -129,67 +129,125 @@ coffee_price(:latte, :venti, 3, false)
 coffee_price(:drip, :tall, 0, true)
 ```
 
-There are a few problems with this:
-- The caller has to remember the order of the arguments
-- When reading the call, there's no way to tell which argument is which
-- The caller must provide values for all arguments, even for the "standard" way of doing things (i.e. no extra shots)
+Having so many parameters on one method introduces several issues:
+1. The caller must remember the order of the arguments
+1. When reading the call, there's no way to tell which argument is which
+1. The caller must provide values for all arguments, even for the "standard" way of doing things (e.g. no extra shots)
 
 All of these put the onus on the caller to do things right, which is just asking for trouble. **A well-written method is easy to use right and hard to use wrong** - by that standard, this is not good code.
 
-### Keyword Arguments
+To address these issues, we will introduce two new concepts: optional arguments and keyword arguments. We'll use each by itself first, then show how they can be used together to create clean, readable code.
+
+### Optional Arguments
+
+We're going to tackle #3 on the above list first by introducing ruby's support of **Optional Arguments**
 
-One way to address these problems is with a technique called _keyword arguments_. Here's what they look like:
+Optional arguments enable you to set default values when defining a function so that the caller doesn't need to pass them unless the value varies from the default. 
 
 ```ruby
 # Define the method
-def coffee_price(type, size, extra_shots: 0, cold: false)
+def coffee_price(type, size, extra_shots = 0, cold = false)
   # ... calculate price, same code as before ...
   return price
 end
 
 # Price for a venti latte with 3 extra shots
-coffee_price(:latte, :venti, extra_shots: 3, cold: false)
+coffee_price(:latte, :venti, 3)
 
 # Price for a tall iced drip
-coffee_price(:drip, :tall, cold: true)
+coffee_price(:drip, :tall, 0, true)
 
 # Price for a grande cappuccino
 coffee_price(:cappuccino, :grande)
 ```
 
-The only things we've changed are the parameters in the method signature, the body of the method remains the same. The end result should resemble [this gist](https://gist.github.com/droberts-ada/07714ad0e2a3bc5a60c50d6a78e02f15).
+Notice that not all the arguments in the method above are optional. Optional Arguments can co-exist with Positional Arguments (and often do).
+Positional arguments must come first, both in the definition and when the method is called.
```

Comment:
Have we defined "positional argument" at this point?

Reply:
Good point! I'll update for that.